### ISSUE TITLE
`terminus build:project:create` fails with composer error

### ISSUE THREAD

**Issue by maskedjellybean** [pantheon-systems/terminus-build-tools-plugin#428]
Running `terminus build:project:create --git=github --team='My Team' d9 pantheon-build-tools-test` fails with this error:

```
Creating a "pantheon-upstreams/drupal-project" project at "./pantheon-build-tools-test"
Installing pantheon-upstreams/drupal-project (dev-master bea9a7abc55c9a8378309e6aa858602b2685cc40)
  - Installing pantheon-upstreams/drupal-project (dev-master bea9a7a): Extracting archive
Created project in /private/var/folders/34/vfkvf9j96xz5b2fmn8ntzpn80000gn/T/local-sitevhMjua/pantheon-build-tools-test
Loading composer repositories with package information
Updating dependencies
Unable to find a compatible set of packages based on your non-dev requirements alone.
Your requirements can be resolved successfully when require-dev packages are present.
You may need to move packages from require-dev or some of their dependencies to require.

  Problem 1
    - Root composer.json requires pantheon-upstreams/upstream-configuration dev-master, found pantheon-upstreams/upstream-configuration[dev-main] but it does not match the constraint. Perhaps dev-master was renamed to dev-main?

 [error]  Command `composer create-project --working-dir=/private/var/folders/34/vfkvf9j96xz5b2fmn8ntzpn80000gn/T/local-sitevhMjua  --repository="{\"url\": \"git@github.com:pantheon-upstreams/drupal-recommended.git\", \"type\": \"vcs\"}" pantheon-upstreams/drupal-project pantheon-build-tools-test -n --stability dev` failed with exit code 2 
```

Here are my versions:
PHP: 7.4
Composer: 2.2.5
Terminus: 3.04

Let me know what other information I can provide to help troubleshoot. 

I was following the documentation at https://pantheon.io/docs/guides/build-tools/create-project/ as well as this repo README (because the documentation is unclear whether the `--team` flag should be your GitHub team or your Pantheon team). Unrelated, but it would also be nice if the documentation clarified that it really does not matter where you are when you run the command, because a local repo is not going to be created.

**Comment by kporras07** [comment]
Hi,

Thanks for reporting this. We're working on this in #425. This was introduced on composer 2.2. So, as a workaround in the meantime, you can rollback to composer 2.1.9 and it should work. Also, if you want to help testing the PR that would be great!

To do so, you should uninstall the current plugin (self:plugin:uninstall), then clone the repo and checkout the branch and in that folder run `terminus self:plugin:install .`

**Comment by kporras07** [comment]
Thanks for reporting. This has been included in 3.0.1 release. Closing this ticket. Feel free open a new ticket if you have further questions/issues
